src/mscorlib/shared/System/String.Comparison.cs

```diff
@@ -761,9 +761,16 @@ public bool Equals(string value, StringComparison comparisonType)
                     if (this.Length != value.Length)
                         return false;
 #if CORECLR
-                    // If both strings are ASCII strings, we can take the fast path.
-                    if (this.IsAscii() && value.IsAscii())
+                    var thisIsAscii = this.IsAscii();
+                    var valueIsAscii = value.IsAscii();
+                    if (thisIsAscii != valueIsAscii)
+                    {
+                        // If one string is ASCII and other is not; then they are not equal
+                        return false;
+                    }
+                    else if (thisIsAscii == true && thisIsAscii == valueIsAscii)
```

Comment:
At this moment we know that `thisIsAscii == valueIsAscii`, so that part of the condition is unnecessary.

Reply:
Changed; though Jit did fix it up to be the same